File: Src/Icm.ContextConsole.Example/MathContext.vb
Imports Icm.ContextConsole

Public Class MathContext
    Inherits BaseContext

    Public Sub Add()
        Dim num1 = Interactor.AskInteger("Number 1")
        Dim num2 = Interactor.AskInteger("Number 2")

        Interactor.ShowMessage(String.Format("{0} + {1} = {2}", num1, num2, num1 + num2))
    End Sub

    Public Sub Multiply()
        Dim num1 = Interactor.AskInteger("Number 1")
        Dim num2 = Interactor.AskInteger("Number 2")

        Interactor.ShowMessage(String.Format("{0} * {1} = {2}", num1, num2, num1 * num2))
    End Sub

End Class

File: Src/Icm.ContextConsole.Example/MathContext.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using Icm.ContextConsole;

public class MathContext : BaseContext
{

	public string PostFix { get; set; }

	public override string Name()
	{
		return base.Name() + PostFix;
	}

	public void Add()
	{
		dynamic num1 = Interactor.AskInteger("Number 1");
		dynamic num2 = Interactor.AskInteger("Number 2");

		Interactor.ShowMessage(string.Format("{0} + {1} = {2}", num1, num2, num1 + num2));
	}

	public void Multiply()
	{
		dynamic num1 = Interactor.AskInteger("Number 1");
		dynamic num2 = Interactor.AskInteger("Number 2");

		Interactor.ShowMessage(string.Format("{0} * {1} = {2}", num1, num2, num1 * num2));
	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
